solr/core/src/java/org/apache/solr/cluster/placement/ReplicaMetric.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.cluster.placement;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+/**
+ * Replica metric wrapper that defines a short symbolic name of the metric, the corresponding
+ * internal metric name (as reported in <code>solr.core.[collection].[replica]</code> registry)
+ * and the desired format/unit conversion.
+ */
+public class ReplicaMetric<T> {
```

Comment:
Now I'm confused between `ReplicaMetric` and `ReplicaMetrics`.... So take previous comments with a grain of salt I might have gotten things wrong.

Reply:
`ReplicaMetric` (singular) is a pseudo-enum that encapsulates both the metric name that we want to use (usually something simple), the internal Solr metrics name (which is usually longer and more complicated) and the optional unit/type conversion for the value retrieved from the metrics API.